client/client.go

```diff
@@ -50,19 +52,18 @@ type Client struct {
 	server neoFSAPIServer
 }
 
-var errNonNeoSigner = fmt.Errorf("%w: expected ECDSA_DETERMINISTIC_SHA256 scheme", neofscrypto.ErrIncorrectSigner)
-
 // New creates an instance of Client initialized with the given parameters.
 //
 // See docs of [PrmInit] methods for details. See also [Client.Dial]/[Client.Close].
-//
-// Returned errors:
-//   - [neofscrypto.ErrIncorrectSigner]
 func New(prm PrmInit) (*Client, error) {
 	var c = new(Client)
-	if prm.signer != nil && prm.signer.Scheme() != neofscrypto.ECDSA_DETERMINISTIC_SHA256 {
-		return nil, errNonNeoSigner
+	pk, err := keys.NewPrivateKey()
+	if err != nil {
+		return nil, fmt.Errorf("private key: %w", err)
 	}
+
+	prm.signer = neofsecdsa.SignerRFC6979(pk.PrivateKey)
```

Comment:
with this we can get rid of nil-check in methods like `NetworkInfo`, right? They will never return missing signing error now

Reply:
Removed Signer check inside these methods